src/tools/illink/src/ILLink.Tasks/LinkTask.cs

```diff
@@ -261,7 +261,9 @@ public class ILLink : ToolTask
 				if (!String.IsNullOrEmpty (_illinkPath))
 					return _illinkPath;
 
+#pragma warning disable IL3000 // 'Assembly.Location' always returns an empty string
 				var taskDirectory = Path.GetDirectoryName (Assembly.GetExecutingAssembly ().Location);
+#pragma warning restore IL3000 // 'Assembly.Location' always returns an empty string
 				// The linker always runs on .NET Core, even when using desktop MSBuild to host ILLink.Tasks.
 				_illinkPath = Path.Combine (Path.GetDirectoryName (taskDirectory), "net7.0", "illink.dll");
```

Comment:
Isn't this changing the path? The previous string ended with "illink.dll" but with your suggestion, it would just point to the parent directory. Is that expected?

Reply:
fixed